src/liballoc/Cargo.toml

```diff
@@ -9,6 +9,7 @@ path = "lib.rs"
 
 [dependencies]
 core = { path = "../libcore" }
+libc = { path = "../rustc/libc_shim" }
```

Comment:
liballoc should not depend on libc. This will break code that runs in kernel-like environments where you have a global memory allocator but no libc.

Reply:
What if libc was only a dependency for `cfg(any(unix, target_os = "redox"))`? It’s only used there.

Reply:
I would prefer if liballoc was completely OS-independent. I am currently using Rust to make Linux binaries which do not link to libc (just using raw syscalls).

This is why, when I originally implemented printing a message on OOM (#30801), I allowed the OOM to be dynamically set at runtime. It would default to `intrinsics::abort`, but would be overridden by libstd on startup to a handler that prints a message.

Reply:
So you’re saying that a `oom` method should be added to the `GlobalAlloc` trait so that `#[global_allocator]` can forward it?

Reply:
I was thinking of adding back a dynamic OOM handler, but your idea is better.

Reply:
Never mind, I see later in the thread that you mention `set_oom_handler`.